src/test/java/org/sagebionetworks/bridge/exporter/integration/ExportTest.java

```diff
@@ -0,0 +1,813 @@
+
+    @Test
+    public void testDailyIgnoreLastExportDateTime() throws Exception {
+        // upload a test record to create last export date time field
+        assertExport(null, ExportType.DAILY, uploadId, 0, false, false, false);
+
+        // modify uploadedOn field in record table to a fake datetime earlier than 24 hours ago
+        // so that the first upload will not be exported again by exporter
+        UploadValidationStatus
+                uploadStatus = user.getClient(ForConsentedUsersApi.class).getUploadStatus(uploadId).execute().body();
+
+        ddbRecordTable.updateItem("id", uploadStatus.getRecord().getId(),
+                new AttributeUpdate("uploadedOn").put(DateTime.now().minusDays(1).minusHours(1).getMillis()));
+
+        // then export another upload
+        createAndEncryptTestFiles(TEST_FILES_A_2, TEST_FILES_B_2);
+        String uploadId2 = testUpload("./legacy-survey-encrypted"); // using the same file name
+
+        // then upload with ignore last export date time
+        assertExport(null, ExportType.DAILY, uploadId2, 1, false, true, false);
+    }
+
+    @Test
+    public void testHourlyIgnoreLastExportDateTime() throws Exception {
+        assertExport(null, ExportType.HOURLY, uploadId, 0, false, false, false);
+
+        UploadValidationStatus
+                uploadStatus = user.getClient(ForConsentedUsersApi.class).getUploadStatus(uploadId).execute().body();
+
+        ddbRecordTable.updateItem("id", uploadStatus.getRecord().getId(),
+                new AttributeUpdate("uploadedOn").put(DateTime.now().minusHours(2).getMillis()));
+
+        // then export another upload
+        createAndEncryptTestFiles(TEST_FILES_A_2, TEST_FILES_B_2);
+        String uploadId2 = testUpload("./legacy-survey-encrypted"); // using the same file name
+
+        // check again
+        assertExport(null, ExportType.HOURLY, uploadId2, 1, false, true, false);
+    }
+
+    @Test
+    public void testEndDateTimeBeforeLastExportDateTime() throws Exception {
+        DateTime dateTimeBeforeExport = DateTime.now();
+        BridgeExporterRequest exporterRequest = new BridgeExporterRequest.Builder().withExportType(ExportType.HOURLY)
+                .withEndDateTime(dateTimeBeforeExport).withStudyWhitelist(ImmutableSet.of(studyId)).build();
+
+        // the first one should be normal
+        assertExport(exporterRequest, null, uploadId, 0, false, false, false);
+
+        // the second one will not export anything -- synapse table will remain the same
+        assertExport(exporterRequest, null, uploadId, 0, false, false, false);
+
+        // then test endDateTime before lastExportDateTime -- there should be no change as well
+        exporterRequest = new BridgeExporterRequest.Builder().withExportType(ExportType.HOURLY)
+                .withEndDateTime(dateTimeBeforeExport.minusHours(1)).withStudyWhitelist(ImmutableSet.of(studyId)).build();
+
+        sqsHelper.sendMessageAsJson(EXPORTER_SQS_URL, exporterRequest, 5);
```

Comment:
oh the second one is for case of endDateTime is equal to lastExportDateTime -- no effect on synapse table; the third one is for case of endDateTime before lastExportDateTime